iroh/src/lib.rs

```diff
@@ -249,6 +249,7 @@ pub mod discovery;
 pub mod dns;
 pub mod endpoint;
 pub mod metrics;
+pub mod net_report;
```

Comment:
It'd be nice to hide this behind a `unstable_api` cargo feature (or whatever bikeshed colour you like)?  Apparently @dignifiedquire had opinions on something like this in the past though, I don't know if favour or against such constructs.  Or maybe keep it entirely internal, as the API for doctor will probably become something else now according to the discussion on discourse?

Reply:
lets merge this as is, and get the API reduction surface before the next release in as well would be my suggestion

Reply:
After our planning meeting, we decided I'm going to reduce the API surface to just what is necessary for `doctor` and get this merged by Friday.